### DISCUSSION TITLE
Programmatically connecting to a kernel

### DISCUSSION THREAD

**Discussion by vadimcn** [microsoft/vscode-jupyter#13459] [Questions and Answers]
Can another extension programmatically instruct the Jupyter extension to create a new notebook (or open an existing one) and connect it to a running kernel (by providing its address)?

**Comment by DonJayamanne** [comment]
Here's the API https://github.com/microsoft/vscode-jupyter/issues/13251

**Reply by DonJayamanne** [reply]
Also check out the API here to get a list of the existing kernels and the like https://github.com/microsoft/vscode-jupyter/blob/armBuilds/src/standalone/api/extension.d.ts#L335
If want to connect to an existing kernel with an address then you need to use the `registerRemoteServerProvider` and return `IJupyterUriProvider`
